src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsApplication.cs

```diff
@@ -164,6 +165,24 @@ private WebApplication CreateApplication()
             configurer(app);
         }
 
+        app.Lifetime.ApplicationStarted.Register(
+            () =>
+            {
+                // NOTE:
+                //
+                // In Kubernetes, the creator of the socket file (this pluggable component) will not be the same user
+                // as the reader/writer of the socket file (the Dapr sidecar), unlike when running the component
+                // locally. Therefore, once the socket file has been created (after start), the permissions need be
+                // updated to allow global read/write.
+
+                var fileInfo = new UnixFileInfo(socketPath);
+
+                fileInfo.FileAccessPermissions =
+                    FileAccessPermissions.UserRead | FileAccessPermissions.UserWrite
+                    | FileAccessPermissions.GroupRead | FileAccessPermissions.GroupWrite
+                    | FileAccessPermissions.OtherRead | FileAccessPermissions.OtherWrite;
+            });
```

Comment:
There are 2 ways to address this issue. One is like you mention: fixing the permission after the creation. I don't know how it is in .Net but in Java handling this presents a race condition between the component start and daprd start. 

The other option is using `umask` to set the default permission for any file created by the process. This one is not subject to that race condition _but_ it is a Unix-only syscall. IIRC this is exposed in .Net without extra packets or APIs and would work in Linux and Mac -- the target platforms.

Reply:
The permissions update happens immediately after application start, so that *could* be a race.  I can look to see if there's a hook that's closer to creation of the socket file but before the application is ready to serve requests.  I don't know how I feel about `umask`, given we don't know what files (and where) a pluggable component might create, and developers might not know what files their own dependencies create.

Reply:
Thinking a bit more about this, won't there *always* be a potential race?  Even if the socket file is created with the permissions needed by the sidecar, that doesn't mean the application is ready to receive requests on that socket.  There is also logic in the application to remove any previous socket file; that could also potentially happen after the sidecar has already noticed it and attempted to connect.  It feels like the sidecar will always need some sort of retry logic to accommodate the various ways in which a pluggable component may not yet be ready for use.